pkgs/development/interpreters/python/mk-python-derivation.nix

```diff
@@ -104,6 +104,49 @@ else
 let
   inherit (python) stdenv;
 
+  validatePythonMatches = attrName: let
+    isPythonModule = drv:
+      # all pythonModules have the pythonModule attribute
+      (drv ? "pythonModule")
+      # Some pythonModules are turned in to a pythonApplication by setting the field to false
+      && (!builtins.isBool drv.pythonModule);
```

Comment:
in the future we can consider a second passthru attribute so you can do this test always.

Reply:
I was just going to call out this condition and check to see if you thought it was the right condition!